app/main.py

```diff
@@ -2,0 +1,39 @@
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False,
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        if health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden:
+            self.hidden = False
+        else:
+            self.hidden = True
+
+
+class Carnivore(Animal):
+    def bite(self, animal: Animal) -> None:
+        if not isinstance(animal, Carnivore) and animal.hidden is False:
+            damage = 50
+            animal.health = animal.health - damage
+            if animal.health <= 0:
+                Animal.alive.remove(animal)
```

Comment:
In the `bite` method, there is no check to ensure that the `animal` is alive before attempting to bite it. Consider adding a condition to check if `animal.health > 0` before applying damage and removing it from `Animal.alive`.